### DISCUSSION TITLE
Promote jump statements syntactically to expressions, similar to how `throw` expressions work.

### DISCUSSION THREAD

**Discussion by Kile-Asmussen** [dotnet/csharplang#7754] [Language Ideas]
This is, in my experience a commonly-used pattern:

```csharp
var foo = bar.Baz() ?? throw new Exception("message"):
```

`throw` is a jump statement, and therefore its use as an expression is well-founded: it will never reach the end-point, and therefore whatever value evaluations may come after, are irrelevant.

My proposal is that we allow the same of at least `return`.

It would maybe be beneficial to allow it of `yield break` as well, due to similar usecase for early termination. The utility of allowing it for `yield return` is quiestionable, as is allowing it for `break` and `continue`. The less said about `goto` the better.

The following is another extremely common pattern:

```csharp
if (arg is null)
  return default;
```

or the more terse version that explicitly matches `arg` into a non-null `val` to appease the type checker

```csharp
if (arg is not {} val)
  return default;
```

which could then be abbreviated to

```csharp
var val = arg ?? return default;
```

and even things like

```csharp
arg ??= return default;
```

would be possible.

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/867.
